src/pages/Staking/index.tsx

```diff
@@ -1,10 +1,723 @@
+    const lockTimeWithFallback = data[LOCK_TIME] || '0'; // Weeks
+
+    const monetaryAmount = newMonetaryAmount(lockingAmountWithFallback, GOVERNANCE_TOKEN, true);
+
+    const unlockHeight = currentBlockNumber + getLockBlocks(parseInt(lockTimeWithFallback));
+
+    if (votingBalanceGreaterThanZero) {
+      moreStakeMutation.mutate({
+        amount: monetaryAmount,
+        unlockHeight
+      });
+    } else {
+      initialStakeMutation.mutate({
+        amount: monetaryAmount,
+        unlockHeight
+      });
+    }
+  };
+
+  const validateLockingAmount = (value: string): string | undefined => {
+    const valueWithFallback = value || '0';
+    const monetaryLockingAmount = newMonetaryAmount(valueWithFallback, GOVERNANCE_TOKEN, true);
+
+    if (
+      !extendLockTimeSet &&
+      monetaryLockingAmount.lte(ZERO_GOVERNANCE_TOKEN_AMOUNT)
+    ) {
+      return 'Locking amount must be greater than zero!';
+    }
+
+    if (monetaryLockingAmount.gt(governanceTokenBalance)) {
+      return 'Locking amount must be less than governance token balance!';
+    }
+
+    const planckLockingAmount = monetaryLockingAmount.to.Planck();
+    const lockBlocks = getLockBlocks(parseInt(lockTime));
+    // This is related to the on-chain implementation where currency values are integers.
+    // So less tokens than the period would likely round to 0.
+    // So on the UI, as long as you require more planck to be locked than the number of blocks the user locks for,
+    // it should be good.
+    if (
+      !extendLockTimeSet &&
+      planckLockingAmount.lte(Big(lockBlocks))
+    ) {
+      return 'Planck to be locked must be greater than the number of blocks you lock for!';
+    }
+
+    return undefined;
+  };
+
+  const validateLockTime = (value: string): string | undefined => {
+    const valueWithFallback = value || '0';
+    const numericValue = parseInt(valueWithFallback);
+
+    if (votingBalanceGreaterThanZero && numericValue === 0) {
+      return undefined;
+    }
+
+    if (
+      numericValue < STAKE_LOCK_TIME.MIN ||
+      numericValue > STAKE_LOCK_TIME.MAX
+    ) {
+      return `Please enter a number between ${STAKE_LOCK_TIME.MIN}-${STAKE_LOCK_TIME.MAX}.`;
+    }
+
+    return undefined;
+  };
+
+  const renderVoteStakedAmountLabel = () => {
+    if (
+      voteGovernanceTokenBalanceIdle ||
+      voteGovernanceTokenBalanceLoading
+    ) {
+      return '-';
+    }
+    if (voteGovernanceTokenBalance === undefined) {
+      throw new Error('Something went wrong!');
+    }
+
+    return displayMonetaryAmount(voteGovernanceTokenBalance);
+  };
+
+  const renderRewardAmountLabel = () => {
+    if (
+      rewardAmountAndAPYIdle ||
+      rewardAmountAndAPYLoading
+    ) {
+      return '-';
+    }
+    if (rewardAmountAndAPY === undefined) {
+      throw new Error('Something went wrong!');
+    }
+
+    return displayMonetaryAmount(rewardAmountAndAPY.amount);
+  };
+
+  const renderStakedAmountLabel = () => {
```

Comment:
minor: as above, you could use a ternary here since you're only checking against a single condition.

Reply:
Done!